src/main/resources/Office365/Groovy.xml

```diff
@@ -227,7 +227,13 @@ public class AzureAuth {
    }
 
    def currentURL = URLEncoder.encode(REDIRECT_URI, "UTF-8")
-   def redirectUrl = "${authority}/${tenant}/oauth2/authorize?response_type=code%20id_token&amp;scope= Files.ReadWrites%20User.Read%20User.ReadWriteUser.ReadBasic.All%20Files.ReadWrite%20Files.ReadWrite.All%20Files.ReadWrite.AppFolder%20Sites.ReadWrite.All&amp;response_mode=form_post&amp;redirect_uri=${currentURL}&amp;client_id=${azureClientID}&amp;nonce=${UUID.randomUUID()}";
+   def redirectUrl = "${authority}/${tenant}/oauth2/authorize?" +
+               "response_type=code%20id_token&amp;" +
+               "scope= Files.ReadWrites%20User.Read%20User.ReadWriteUser.ReadBasic.All" +
+               "%20Files.ReadWrite%20Files.ReadWrite.All%20Files.ReadWrite.AppFolder" +
+               "%20Sites.ReadWrite.All&amp;response_mode=form_post&amp;" +
+               "redirect_uri=${currentURL}&amp;client_id=${azureClientID}&amp;" +
+               "nonce=${UUID.randomUUID()}";
```

Comment:
I would rather suggest to use a more controllable structure to express this. Here's what is missing from here:
1. A map to store the parameters
2. A small block of code to convert the map into a queryString
3. Properly encoding. `URLEncoder.encode(REDIRECT_URI, "UTF-8")` above is a good example.
`code%20id_token` is not a good option
4. Append the queryString to the redirectUrl
You will be surprised how nice and maintainable will be the code after this refactoring

Reply:
That took me a while to adjust.
Now addressed in https://github.com/xwikisas/application-office365/pull/11/commits/4aa6cf8aedd9691a3d13a689797843460eea3a02 .
thanks for the suggestion.